pkg/backend/httpstate/backend.go

```diff
@@ -1177,18 +1178,47 @@ func (b *cloudBackend) ExportDeploymentForVersion(
 		return nil, errors.Errorf("%q is not a valid stack version. It should be a positive integer.", version)
 	}
 
-	return b.exportDeployment(ctx, stack.Ref(), &versionNumber)
+	return b.exportDeployment(ctx, stack, &versionNumber, showSecrets)
 }
 
 // exportDeployment exports the checkpoint file for a stack, optionally getting a previous version.
 func (b *cloudBackend) exportDeployment(
-	ctx context.Context, stackRef backend.StackReference, version *int) (*apitype.UntypedDeployment, error) {
+	ctx context.Context, stk backend.Stack, version *int, showSecrets bool) (*apitype.UntypedDeployment, error) {
+
+	stackRef := stk.Ref()
+	if showSecrets {
+		// get the snapshot
+		snap, err := b.getSnapshot(ctx, stk)
+		if err != nil {
+			return nil, err
+		}
+
+		if snap == nil {
+			snap = deploy.NewSnapshot(deploy.Manifest{}, nil, nil, nil)
+		}
+
+		sdep, err := stack.SerializeDeployment(snap, snap.SecretsManager, showSecrets)
+		if err != nil {
+			return nil, errors.Wrap(err, "serializing deployment")
+		}
+
+		data, err := json.Marshal(sdep)
+		if err != nil {
+			return nil, err
+		}
+
+		return &apitype.UntypedDeployment{
+			Version:    3,
+			Deployment: json.RawMessage(data),
+		}, nil
```

Comment:
This doesn't feel quite right - it feels like this codepath should be more similar to the non-showSecrets path - and that the differences should be more clear in another layer - ideally a higher layer so this can be reused across backends, but if not then maybe even a lower layer?

Reply:
As above, we can move this further up but it was done to try and make this as less disruptive as possible

Reply:
I'd prefer that we do this up the stack as well.